code/modules/power/power.dm

```diff
@@ -71,289 +81,197 @@
+			var/obj/machinery/power/M = P
+			found_machines |= M //we wait until the powernet is fully propagates to connect the machines
 
-	//null the powernet reference of all cables & nodes in this powernet
-	var/i=1
-	while(i<=cables.len)
-		var/obj/structure/cable/Cable = cables[i]
-		if(Cable)
-			Cable.powernet = null
-			if(Cable == C)
-				cables.Cut(i,i+1)
-				continue
-		i++
-	i=1
-	while(i<=nodes.len)
-		var/obj/machinery/power/Node = nodes[i]
-		if(Node)
-			Node.powernet = null
-		i++
-
-	// remove the cut cable from the network
-//	C.netnum = -1
-
-	C.loc = null
-
-	powernet_nextlink(P1[1], src)		// propagate network from 1st side of cable, using current netnum	//TODO?
-
-	// now test to see if propagation reached to the other side
-	// if so, then there's a loop in the network
-	var/notlooped = 0
-	for(var/O in P2)
-		if( istype(O, /obj/machinery/power) )
-			var/obj/machinery/power/Machine = O
-			if(Machine.powernet != src)
-				notlooped = 1
-				break
-		else if( istype(O, /obj/structure/cable) )
-			var/obj/structure/cable/Cable = O
-			if(Cable.powernet != src)
-				notlooped = 1
-				break
-
-	if(notlooped)
-		// not looped, so make a new powernet
-		var/datum/powernet/PN = new()
-		powernets += PN
-
-//		if(Debug) world.log << "Was not looped: spliting PN#[number] ([cables.len];[nodes.len])"
-
-		i=1
-		while(i<=cables.len)
-			var/obj/structure/cable/Cable = cables[i]
-			if(Cable && !Cable.powernet)	// non-connected cables will have powernet=null, since they weren't reached by propagation
-				Cable.powernet = PN
-				cables.Cut(i,i+1)	// remove from old network & add to new one
-				PN.cables += Cable
-				continue
-			i++
-
-		i=1
-		while(i<=nodes.len)
-			var/obj/machinery/power/Node = nodes[i]
-			if(Node && !Node.powernet)
-				Node.powernet = PN
-				nodes.Cut(i,i+1)
-				PN.nodes[Node] = Node
-				continue
-			i++
-
-	// Disconnect machines connected to nodes
-	if(node)
-		for(var/obj/machinery/power/P in T1)
-			if(P.powernet && !P.powernet.nodes[src])
-				P.disconnect_from_network()
-//		if(Debug)
-//			world.log << "Old PN#[number] : ([cables.len];[nodes.len])"
-//			world.log << "New PN#[PN.number] : ([PN.cables.len];[PN.nodes.len])"
-//
-//	else
-//		if(Debug)
-//			world.log << "Was looped."
-//		//there is a loop, so nothing to be done
-//		return
-
-
-
-/datum/powernet/proc/reset()
-
-	//see if there's a surplus of power remaining in the powernet and stores unused power in the SMES
-	netexcess = avail - load
-
-	if(netexcess > 100 && nodes && nodes.len)		// if there was excess power last cycle
-		for(var/obj/machinery/power/smes/S in nodes)	// find the SMESes in the network
-			S.restore()				// and restore some of the power that was used
-
-	//updates the viewed load (as seen on power computers)
-	viewload = 0.8*viewload + 0.2*load
-	viewload = round(viewload)
-
-	//reset the powernet
-	load = 0
-	avail = newavail
-	newavail = 0
-
-/datum/powernet/proc/get_electrocute_damage()

```

Comment:
The scaling of powernet damage here should probably be ported over to this new system; these numbers (and effects) were selected for consistency with powergloves and for the arc-flashing effects.